### ISSUE TITLE
Loky cache should be disabled by default

### ISSUE THREAD

**Issue by amogil** [joblib/joblib#836]
Joblib uses loky as a backend by default. Loky reuses previously forked processes during consequent runs. This may lead to unobvious hard to track bugs. For example:

```
import joblib
from joblib import Parallel, delayed\

print(f"Joblib version: {joblib.__version__}")

def process(i):
    return f"{chunk} - {i}"

for chunk in ['chunk1', 'chunk2', 'chunk3']:
    print(f"----------- {chunk} -----------")
    result = Parallel(n_jobs=-1)(delayed(process)(i) for i in range(5))
    print(result)
```

produces

```
Joblib version: 0.13.0
----------- chunk1 -----------
['chunk1 - 0', 'chunk1 - 1', 'chunk1 - 2', 'chunk1 - 3', 'chunk1 - 4']
----------- chunk2 -----------
['chunk1 - 0', 'chunk1 - 1', 'chunk1 - 2', 'chunk1 - 3', 'chunk2 - 4']
----------- chunk3 -----------
['chunk1 - 0', 'chunk1 - 1', 'chunk3 - 2', 'chunk1 - 3', 'chunk1 - 4']
```

Please note that only the "chunk1" string was printed.

In contrast, when 'multiprocessing' backend is being used:

```
import joblib
from joblib import Parallel, delayed\

print(f"Joblib version: {joblib.__version__}")

def process(i):
    return f"{chunk} - {i}"

for chunk in ['chunk1', 'chunk2', 'chunk3']:
    print(f"----------- {chunk} -----------")
    result = Parallel(n_jobs=-1, backend='multiprocessing')(delayed(process)(i) for i in range(5))
    print(result)
```

gives

```
Joblib version: 0.13.0
----------- chunk1 -----------
['chunk1 - 0', 'chunk1 - 1', 'chunk1 - 2', 'chunk1 - 3', 'chunk1 - 4']
----------- chunk2 -----------
['chunk2 - 0', 'chunk2 - 1', 'chunk2 - 2', 'chunk2 - 3', 'chunk2 - 4']
----------- chunk3 -----------
['chunk3 - 0', 'chunk3 - 1', 'chunk3 - 2', 'chunk3 - 3', 'chunk3 - 4']
```

It works as expected.

Loky doesn't spawn processes again and reuses them from the previous run. The `chunk` variable and `for cycle` here is just a synthetic example. It could be everything from any read-only data from the master process to libraries' imports. Consequent joblib's usages can happen in different parts of an app and can lead to really hard to track bugs.

As far as I understand, we can't control this behavior using joblib options, it's joblib implementation details which are not seen to joblib users.

I think it would be better to disable loky process cache by default.

**Comment by ogrisel** [comment]
The bug is caused by a change in cloudpickle. We will fix it as part of #240.

**Comment by ogrisel** [comment]
I confirm that the fix in cloudpipe/cloudpickle#240 fixes this issue.

**Comment by ogrisel** [comment]
I confirm that the fix in cloudpipe/cloudpickle#240 fixes this issue.
